plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/related-by-control.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { CheckboxControl, PanelBody } from '@wordpress/components';
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Internal dependencies
+ */
+import { QueryControlProps, RelatedBy, CoreFilterNames } from '../../types';
+
+const RelatedByControl = ( {
+	query,
+	setQueryAttribute,
+	trackInteraction,
+}: QueryControlProps ) => {
+	const relatedBy = query?.relatedBy as RelatedBy;
+
+	const handleRelatedByChange = (
+		value: boolean,
+		type: 'categories' | 'tags'
+	) => {
+		const newRelatedBy = {
+			...relatedBy,
+			[ type ]: value,
+		};
+
+		setQueryAttribute( {
+			relatedBy: newRelatedBy,
+		} );
+
+		trackInteraction( CoreFilterNames.RELATED_BY );
+
+		recordEvent(
+			'blocks_product_collection_inspector_control_related_by_changed',
+			{
+				categories: newRelatedBy?.categories || false,
+				tags: newRelatedBy?.tags || false,
+			}
+		);
+	};
+
+	return (
+		<PanelBody title={ __( 'Relate by', 'woocommerce' ) }>
```

Comment:
I saw designs have "Relate by" but I guess it's a typo:

```suggestion
		<PanelBody title={ __( 'Related by', 'woocommerce' ) }>
```

And if not, code reference should reflect that and we should change `relatedBy` to `relateBy` 😅

Reply:
Good catch! As discussed on Slack, I have made the changes in [3cd67d1](https://github.com/woocommerce/woocommerce/pull/52580/commits/3cd67d1e019116f77b84cfb662a45e3c68f0eacb) 😀